Sources/Memcache/ChannelHandler/Messages/Backend+Flags.swift

```diff
@@ -0,0 +1,45 @@
+import NIOCore
+
+extension MemcacheBackendMessage {
+    struct Flags: MemcacheMessagePayloadDecodable, Equatable, ExpressibleByArrayLiteral {
+        let flags: [String] // TODO: Do we want something like (Character, token: String?) instead? Or a struct?
```

Comment:
If we try to parse the flags a bit further in this stage, we can catch malformed messages earlier and it might make parsing the meta-command-specific flags a bit easier. Flags are always a single character followed by an optional token string. Some tokens have a length limit but I don't think it makes sense to try to enforce it in this stage.

Reply:
I think the flags should be an enum with associated values. And I think we should already enforce this here. I think users should be able to never decode anything after this stage and trust the values 100%.

Reply:
Totally agree 👍 One potential issue however is that flags can have slightly different meanings depending on the sent command. For example, the `N(token)` flags is described as [_vivify on miss, takes TTL as argument_ for Meta Get](https://github.com/memcached/memcached/blob/597645db6a2b138710f01ffe5e92e453117b987a/doc/protocol.txt#L557), and as [_auto create item on miss with supplied TTL_  for Meta Arithmetic](https://github.com/memcached/memcached/blob/597645db6a2b138710f01ffe5e92e453117b987a/doc/protocol.txt#L881) commands. I couldn't find any collisions so we just need to choose good names for the enum cases.

One thing we can't verify here though is which flags can be part of a message. This also depends on the sent command.

Reply:
Or do you think we should just be unopinionated and directly use the flag characters as the enum case names?

Reply:
I implemented decoding of flags I think as far as possible and added checks for things like data type and length where it makes sense.